util.py

```diff
@@ -419,6 +421,93 @@ def winedll_override(dll: str, dtype: Literal['n', 'b', 'n,b', 'b,n', '']) -> No
     )
 
 
+def patch_libcuda() -> bool:
+    """Patches libcuda to work around games that crash with DLSS due to wonky memory allocation.
+
+    The patched library is overwritten at every launch and is placed in .cache
+
+    Returns true if the library was patched correctly. Otherwise returns false
+    """
+    cache_dir = os.path.expanduser('~/.cache/protonfixes')
+    os.makedirs(cache_dir, exist_ok=True)
+
+    try:
+        library_base = ctypes.util.find_library('cuda')
+        if not library_base:
+            log.warn('libcuda.so not found on the system.')
+            return False
+
+        library_paths = [
+            '/lib64', '/usr/lib64', '/usr/local/lib64',
+            '/lib', '/usr/lib', '/usr/local/lib'
+        ]
+
+        libcuda_path = None
+        for lib_dir in library_paths:
+            full_path = os.path.join(lib_dir, library_base)
+            if os.path.exists(full_path):
+                # Check if it's a 64-bit ELF file
+                try:
+                    with open(full_path, 'rb') as f:
+                        e_ident = f.read(5)
+                        if e_ident[:4] != b'\x7fELF':
+                            log.info(f'{full_path} is not an ELF file.')
+                            continue  # Not an ELF file
+                        ei_class = e_ident[4]
+                        if ei_class == 2:
+                            # 64-bit ELF
+                            libcuda_path = os.path.abspath(full_path)
+                            break
+                        else:
+                            log.info(f'{full_path} is not a 64-bit ELF file.')
+                            continue  # Not 64-bit ELF
+                except Exception as e:
+                    log.error(f'Error reading {full_path}: {e}')
+                    continue
+
+        if not libcuda_path:
+            log.error(f'libcuda.so found as {library_base}, but not found as a 64-bit library in standard library directories.')
+            return False
+
+        log.info(f'Found 64-bit libcuda.so at: {libcuda_path}')
+
+        patched_library = os.path.join(cache_dir, 'libcuda.patched.so')
+        try:
+            with open(libcuda_path, 'rb') as f:
+                binary_data = f.read()
+        except OSError as e:
+            log.error(f'Unable to read libcuda.so: {e}')
+            return False
+
+        hex_data = binary_data.hex()
+        hex_data = hex_data.replace('000000f8ff000000', '000000f8ffff0000')
+        patched_binary_data = bytes.fromhex(hex_data)
```

Comment:
I think we need a comment and a link so this looks less suspect

Reply:
Indeed, I was going for minimal comments, but without context this definitely can look sketchy. I'll change this.

Reply:
Should be fixed with https://github.com/Open-Wine-Components/umu-protonfixes/pull/136/commits/09785517d12838d0789da6c9c385714b6703f56b

let me know if you want more (or less) information